pkg/webhook/utils/utils.go

```diff
@@ -201,5 +180,69 @@ func ParseNetworkConfigOfPodByPriority(ctx context.Context, c client.Reader, pod
 		}
 	}
 
+	networkType = ipamtypes.ParseNetworkTypeFromString(networkTypeStr)
+	if len(ipFamily) == 0 {
+		ipFamily = ipamtypes.ParseIPFamilyFromString(ipFamilyStr)
+	}
+
+	if len(networkName) > 0 {
+		network := &networkingv1.Network{}
+		if err = c.Get(ctx, types.NamespacedName{Name: networkName}, network); err != nil {
+			err = fmt.Errorf("failed to get network %v: %v", networkName, err)
+			return
+		}
+
+		// specified network takes higher priority than network type defaulting, if no network type specified
+		// from pod, then network type should inherit from network type of specified network from pod
+		if len(networkTypeStr) == 0 {
+			networkType = ipamtypes.ParseNetworkTypeFromString(string(networkingv1.GetNetworkType(network)))
+		}
+
+		networkNodeSelector = network.Spec.NodeSelector
+	}
+
+	if !ipamtypes.IsValidFamilyMode(ipFamily) {
+		err = fmt.Errorf("unrecognized ip family %s", ipFamily)
+		return
+	}
+
+	if !ipamtypes.IsValidNetworkType(networkType) {
+		err = fmt.Errorf("unrecognized network type %s", networkType)
+		return
+	}
+
+	return
+}
+
+func parseNetworkConfigByExistIPInstances(ctx context.Context, c client.Reader, opts ...client.ListOption) (networkName string,
+	ipFamily ipamtypes.IPFamilyMode, err error) {
+	ipList := &networkingv1.IPInstanceList{}
+	if err = c.List(ctx, ipList, opts...); err != nil {
+		return
+	}
+
+	ipFamilyCounter := 0
+	for i := range ipList.Items {
+		// ignore terminating ipInstance
+		if ipList.Items[i].DeletionTimestamp == nil {
+			networkName = ipList.Items[i].Spec.Network
+			ipFamilyCounter++
+		}
+	}
+
+	switch ipFamilyCounter {
+	case 1:
+		if networkingv1.IsIPv6IPInstance(&ipList.Items[0]) {
```

Comment:
should terminating IPInstance be ignored here?

Reply:
done